apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1303,61 +1282,47 @@ where
                     // We know last edge is not a cast.
                     runtime_types_before_tail_if_last_is_cast: None,
                     defer_on_tail: defer,
-                    context_to_selection: self.context_to_selection.clone(),
-                    parameter_to_context: self.parameter_to_context.clone(),
+                    set_context_ids: self.set_context_ids.clone(),
+                    arguments_to_context_usages: self.arguments_to_context_usages.clone(),
                 });
             }
         }
 
-        let (new_edge_conditions, new_context_to_selection, new_parameter_to_context) =
+        let (new_edge_conditions, new_set_context_ids, new_arguments_to_context_usages) =
             self.merge_edge_conditions_with_resolution(&condition_path_tree, &context_map);
-        let last_parameter_to_context = new_parameter_to_context.last();
+        let last_arguments_to_context_usages = new_arguments_to_context_usages.last();
 
-        if let Some(Some(last_parameter_to_context)) = last_parameter_to_context {
+        if let Some(Some(last_arguments_to_context_usages)) = last_arguments_to_context_usages {
             // TODO: Perhaps it is better to explicitly cast this to `GraphPathTriggerRefMut` and
             // pull out the field from there.
             if let Some(field) = trigger.get_field_mut() {
-                let mut schema = field.schema.schema().clone().into_inner();
```

Comment:
Bugs:
- Instead of creating a fake modified supergraph schema (which then also has to be validated), we should be switching the field to use the relevant subgraph schema of the edge, that's what declared the `@fromContext` arguments to start with.